libs/prov/prov_k8s_cortx_deploy.py

```diff
@@ -463,63 +473,73 @@ def update_sol_yaml(self, worker_obj: list, filepath: str, cortx_image: str,
         valid_disk_count = sns_spare + sns_data + sns_parity
-        metadata_devices = []
-        for node_count, node_obj in enumerate(worker_obj, start=1):
-            LOGGER.info(node_count)
-            device_list = node_obj.execute_cmd(cmd=common_cmd.CMD_LIST_DEVICES,
-                                               read_lines=True)[0].split(",")
-            device_list[-1] = device_list[-1].replace("\n", "")
-            metadata_devices = device_list[1:cvg_count + 1]
-            # This will split the metadata disk list
-            # into metadata devices per cvg
-            # 2 is defined the split size based
-            # on disk required for metadata,system
-            device_list_len = len(device_list)
-            new_device_lst_len = (device_list_len - cvg_count - 1)
-            count = cvg_count
-            if data_disk_per_cvg == 0:
-                data_disk_per_cvg = int(len(device_list[cvg_count + 1:]) / cvg_count)
-
-            LOGGER.debug("Data disk per cvg : %s", data_disk_per_cvg)
-            # The condition to validate the config.
-            if not skip_disk_count_check and valid_disk_count > \
-                    (cvg_count * node_list):
-                return False, "The sum of data disks per cvg " \
-                              "is less than N+K+S count"
-
-            if new_device_lst_len < data_disk_per_cvg * cvg_count:
-                return False, "The requested data disk is more than" \
-                              " the data disk available on the system"
-            # This condition validated the total available disk count
-            # and split the disks per cvg.
-            data_devices_f = device_list[cvg_count + 1:]
-            if (data_disk_per_cvg * cvg_count) < new_device_lst_len:
-                count_end = int(data_disk_per_cvg)
-                data_devices.append(data_devices_f[0:count_end])
-                while count:
-                    count = count - 1
-                    new_end = int(count_end + data_disk_per_cvg)
-                    if new_end > new_device_lst_len:
-                        break
-                    data_devices_ad = data_devices_f[count_end:new_end]
-                    count_end = int(count_end + data_disk_per_cvg)
-                    data_devices.append(data_devices_ad)
-            else:
-                LOGGER.debug("Data devices : else : %s", data_devices_f)
-                LOGGER.debug("data disk per cvg : %s", data_disk_per_cvg)
-                data_devices = [data_devices_f[i:i + data_disk_per_cvg]
-                                for i in range(0, len(data_devices_f), data_disk_per_cvg)]
-
-            # Create dict for host and disk
-            system_disk = device_list[0]
-            schema = {node_obj.hostname: system_disk}
-            sys_disk_pernode.update(schema)
-        LOGGER.info("Metadata disk %s", metadata_devices)
-        LOGGER.info("data disk %s", data_devices)
+
+        def _cvg_config(data_disk_per_cvg):
```

Comment:
Same comment as in line 361. Protected can be part of the class itself . Don't you want to reuse it in this module ? Or the functionality is part creating solution yaml itself..